aggregator/entry.go

```diff
@@ -118,8 +172,8 @@ func (e *Entry) AddMetricWithPolicies(
 		return errEntryClosed
 	}
 
-	if e.shouldUpdatePoliciesWithLock(currTime, policies.Version, policies.Cutover) {
-		if err := e.updatePoliciesWithLock(mu.Type, mu.ID, mu.OwnsID, policies.Version, policies.Policies()); err != nil {
+	if e.shouldUpdatePoliciesWithLock(currTime, sp) {
```

Comment:
Do we need to acquire a write lock before we read the value of `e.closed`? Perhaps we can defer it to here before we call `shouldUpdatePoliciesWithLock`?

Also, perhaps we should distinguish the functions which need a read lock from those which need a write lock? For example, `shouldUpdatePoliciesWithLock` is called with both but `updatePoliciesWithLock` needs a lock since it actually updates `e`.